src/AsmResolver.PE/DotNet/VTableFixups/VTableFixupDirectory.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using AsmResolver.IO;
+
+namespace AsmResolver.PE.DotNet.VTableFixups
+{
+    /// <summary>
+    /// Represents the VTable Fixup Directory in the Cor20 header.
+    /// </summary>
+    public class VTableFixupDirectory : Collection<VTableFixup>, ISegment
+    {
+        private readonly Dictionary<VTableFixup, uint> _tableRvas = new();
```

Comment:
The purpose of the dictionary is to store the location of the entries of the table, and the RVA is actually the first field of the structure of VTable Fixups: https://github.com/dotnet/runtime/blob/ec6bec455514070cc69e4120974796338bcb7cb7/src/coreclr/src/inc/corhdr.h#L1272

Reply:
Yes, but this setup forces every implementation of `IPEFileBuilder` to place the actual lists of tokens right after the vtablefixups directory. Spec states that the actual lists of metadata tokens can be _anywhere_ in the PE, as long as it's in a section with read/write access flags (ECMA-335, Section II.25.3.3.3, page 284). Therefore, I think a different `IPEFileBuilder` implementation (such as one that preserves as much as possible of the original PE file) should be able to decide a different location for these vtables.

Perhaps we can let `VTableFixup` implement `ISegment` or inherit from `SegmentBase` instead so that it becomes relocatable by a `IPEFileBuilder` instance?